cannot run on ios

hi. I want to use your FirebaseVisionTextDetector.
my app works fine on android, but when tried to run on ios, build failed with following errors.
what would you suppose wrong with my project?

Xcode's output:
↳
    === BUILD TARGET mlkit OF PROJECT Pods WITH CONFIGURATION Debug ===
    In file included from /Users/kunugisayuri/flutter/.pub-cache/hosted/pub.dartlang.org/mlkit-0.6.0/ios/Classes/MlkitPlugin.m:1:
    In file included from /Users/kunugisayuri/flutter/.pub-cache/hosted/pub.dartlang.org/mlkit-0.6.0/ios/Classes/MlkitPlugin.h:1:
    In file included from /Users/kunugisayuri/StudioProjects/insurance_card_reader/ios/Pods/Headers/Public/Flutter/Flutter/Flutter.h:37:
    In file included from /Users/kunugisayuri/StudioProjects/insurance_card_reader/ios/Pods/Headers/Public/Flutter/Flutter/FlutterAppDelegate.h:11:
    /Users/kunugisayuri/StudioProjects/insurance_card_reader/ios/Pods/Headers/Public/Flutter/Flutter/FlutterPlugin.h:140:58: warning: this block declaration is not a prototype [-Wstrict-prototypes]
                          completionHandler:(nonnull void (^)())completionHandler;
                                                             ^
                                                              void
    /Users/kunugisayuri/flutter/.pub-cache/hosted/pub.dartlang.org/mlkit-0.6.0/ios/Classes/MlkitPlugin.m:25:1: error: unknown type name 'FIRVisionTextDetector'; did you mean 'FIRVisionFaceDetector'?
    FIRVisionTextDetector *textDetector;
    ^~~~~~~~~~~~~~~~~~~~~
    FIRVisionFaceDetector
    In file included from /Users/kunugisayuri/flutter/.pub-cache/hosted/pub.dartlang.org/mlkit-0.6.0/ios/Classes/MlkitPlugin.m:2:
    In file included from /Users/kunugisayuri/StudioProjects/insurance_card_reader/ios/Pods/Headers/Private/Firebase/Firebase.h:62:
    In file included from /Users/kunugisayuri/StudioProjects/insurance_card_reader/ios/Pods/Headers/Public/FirebaseMLVision/FirebaseMLVision/FirebaseMLVision.h:19:
    /Users/kunugisayuri/StudioProjects/insurance_card_reader/ios/Pods/Headers/Public/FirebaseMLVision/FirebaseMLVision/FIRVisionFaceDetector.h:22:12: note: 'FIRVisionFaceDetector' declared here
    @interface FIRVisionFaceDetector : NSObject
               ^
    /Users/kunugisayuri/flutter/.pub-cache/hosted/pub.dartlang.org/mlkit-0.6.0/ios/Classes/MlkitPlugin.m:186:41: error: type argument 'FIRVisionText' must be a pointer (requires a '*')
    NSDictionary *visionTextToDictionary(id<FIRVisionText> visionText) {
                                            ^
                                                          *
    /Users/kunugisayuri/flutter/.pub-cache/hosted/pub.dartlang.org/mlkit-0.6.0/ios/Classes/MlkitPlugin.m:186:38: error: type arguments cannot be applied to non-class type 'id'
    NSDictionary *visionTextToDictionary(id<FIRVisionText> visionText) {
                                         ^ ~~~~~~~~~~~~~~~
    /Users/kunugisayuri/flutter/.pub-cache/hosted/pub.dartlang.org/mlkit-0.6.0/ios/Classes/MlkitPlugin.m:72:32: error: no visible @interface for 'FIRVision' declares the selector 'textDetector'
            textDetector = [vision textDetector];
                            ~~~~~~ ^~~~~~~~~~~~
    /Users/kunugisayuri/flutter/.pub-cache/hosted/pub.dartlang.org/mlkit-0.6.0/ios/Classes/MlkitPlugin.m:82:43: error: type argument 'FIRVisionText' must be a pointer (requires a '*')
                                     for (id <FIRVisionText> feature in features) {
                                              ^
                                                            *
    /Users/kunugisayuri/flutter/.pub-cache/hosted/pub.dartlang.org/mlkit-0.6.0/ios/Classes/MlkitPlugin.m:82:39: error: type arguments cannot be applied to non-class type 'id'
                                     for (id <FIRVisionText> feature in features) {
                                          ^  ~~~~~~~~~~~~~~~
    /Users/kunugisayuri/flutter/.pub-cache/hosted/pub.dartlang.org/mlkit-0.6.0/ios/Classes/MlkitPlugin.m:188:17: error: property 'cornerPoints' not found on object of type '__strong id'
        [visionText.cornerPoints enumerateObjectsUsingBlock:^(NSValue * _Nonnull obj, NSUInteger idx, BOOL * _Nonnull stop) {
                    ^
    /Users/kunugisayuri/flutter/.pub-cache/hosted/pub.dartlang.org/mlkit-0.6.0/ios/Classes/MlkitPlugin.m:195:35: error: property 'text' not found on object of type '__strong id'
                 @"text" : visionText.text,
                                      ^
    1 warning and 8 errors generated.
Could not build the application for the simulator.
Error launching application on iPhone X.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,7 @@
+## 0.6.1
+
+* update ios textDetector implement for firebase API update.
+
 ## 0.6.0
 
 * support custom model interpreter for android (cloud hosting model only).

--- a/example/ios/Runner.xcodeproj/project.pbxproj
+++ b/example/ios/Runner.xcodeproj/project.pbxproj
@@ -248,10 +248,12 @@
 			);
 			inputPaths = (
 				"${SRCROOT}/Pods/Target Support Files/Pods-Runner/Pods-Runner-resources.sh",
+				"${PODS_CONFIGURATION_BUILD_DIR}/GoogleMobileVision/GoogleMVFaceDetectorResources.bundle",
 				"${PODS_CONFIGURATION_BUILD_DIR}/GoogleMobileVision/GoogleMVTextDetectorResources.bundle",
 			);
 			name = "[CP] Copy Pods Resources";
 			outputPaths = (
+				"${TARGET_BUILD_DIR}/${UNLOCALIZED_RESOURCES_FOLDER_PATH}/GoogleMVFaceDetectorResources.bundle",
 				"${TARGET_BUILD_DIR}/${UNLOCALIZED_RESOURCES_FOLDER_PATH}/GoogleMVTextDetectorResources.bundle",
 			);
 			runOnlyForDeploymentPostprocessing = 0;

--- a/ios/Classes/MlkitPlugin.m
+++ b/ios/Classes/MlkitPlugin.m
@@ -22,7 +22,7 @@
                         };
 }
 
-FIRVisionTextDetector *textDetector;
+FIRVisionTextRecognizer *textDetector;
 FIRVisionBarcodeDetector *barcodeDetector;
 FIRVisionFaceDetector *faceDetector;
 FIRVisionLabelDetector *labelDetector;
@@ -69,37 +69,18 @@ UIImage* imageFromImageSourceWithData(NSData *data) {
     FIRVisionImage *image = [[FIRVisionImage alloc] initWithImage:uiImage];
     
     if ([call.method hasPrefix:@"FirebaseVisionTextDetector#detectFrom"]) {
-        textDetector = [vision textDetector];
-        [textDetector detectInImage:image
-                         completion:^(NSArray<FIRVisionText *> *features,
-                                      NSError *error) {
+        textDetector = [vision onDeviceTextRecognizer];
+        [textDetector processImage:image
+                        completion:^(FIRVisionText *_Nullable resultText,
+                                     NSError *_Nullable error) {
                              if (error != nil) {
                                  [ret addObject:error.localizedDescription];
                                  result(ret);
                                  return;
-                             } else if (features != nil) {
+                             } else if (resultText != nil) {
                                  // Recognized text
-                                 for (id <FIRVisionText> feature in features) {
-                                     // Blocks contain lines of text
-                                     if ([feature isKindOfClass:[FIRVisionTextBlock class]]) {
-                                         FIRVisionTextBlock *block = (FIRVisionTextBlock *)feature;
-                                         [ret addObject:visionTextBlockToDictionary(block)];
-                                     }
-                                     
-                                     // Lines contain text elements
-                                     else if ([feature isKindOfClass:[FIRVisionTextLine class]]) {
-                                         FIRVisionTextLine *line = (FIRVisionTextLine *)feature;
-                                         [ret addObject:visionTextLineToDictionary(line)];
-                                     }
-                                     
-                                     // Text elements are typically words
-                                     else if ([feature isKindOfClass:[FIRVisionTextElement class]]) {
-                                         FIRVisionTextElement *element = (FIRVisionTextElement *)feature;
-                                         [ret addObject:visionTextElementToDictionary(element)];
-                                     }
-                                     else {
-                                         [ret addObject:visionTextToDictionary(feature)];
-                                     }
+                                 for (FIRVisionTextBlock *block in resultText.blocks) {
+                                     [ret addObject:visionTextBlockToDictionary(block)];
                                  }
                              }
                              result(ret);
@@ -181,24 +162,6 @@ UIImage* imageFromImageSourceWithData(NSData *data) {
     }else {
         result(FlutterMethodNotImplemented);
     }
-}
-
-NSDictionary *visionTextToDictionary(id<FIRVisionText> visionText) {
-    __block NSMutableArray<NSDictionary *> *points =[NSMutableArray array];
-    [visionText.cornerPoints enumerateObjectsUsingBlock:^(NSValue * _Nonnull obj, NSUInteger idx, BOOL * _Nonnull stop) {
-        [points addObject:@{
-                            @"x": @(((__bridge CGPoint *)obj)->x),
-                            @"y": @(((__bridge CGPoint *)obj)->y),
-                            }];
-    }];
-    return @{
-             @"text" : visionText.text,
-             @"rect_left": @(visionText.frame.origin.x),
-             @"rect_top": @(visionText.frame.origin.y),
-             @"rect_right": @(visionText.frame.origin.x + visionText.frame.size.width),
-             @"rect_bottom": @(visionText.frame.origin.y + visionText.frame.size.height),
-             @"points": points,
-             };
 }
 
 NSDictionary *visionTextBlockToDictionary(FIRVisionTextBlock * visionTextBlock) {

--- a/pubspec.yaml
+++ b/pubspec.yaml
@@ -1,6 +1,6 @@
 name: mlkit
 description: A Flutter plugin to use the Firebase ML Kit.
-version: 0.6.0
+version: 0.6.1
 author: azihsoyn <azihsoyn@gmail.com>
 homepage: https://github.com/azihsoyn/flutter_mlkit
 

